starknet_py/tests/e2e/mock/compile_contracts.sh

```diff
@@ -1,32 +1,81 @@
 #!/bin/bash
-
+set -e
 MOCK_DIRECTORY=starknet_py/tests/e2e/mock
-CONTRACTS_DIRECTORY="$MOCK_DIRECTORY"/contracts
-CONTRACTS_COMPILED_DIRECTORY="$MOCK_DIRECTORY"/contracts_compiled
+CONTRACTS_DIRECTORY_V1="$MOCK_DIRECTORY"/contracts_v1
+CONTRACTS_DIRECTORY_V2="$MOCK_DIRECTORY"/contracts_v2
 
-# delete all artifacts except precompiled ones
-find $CONTRACTS_COMPILED_DIRECTORY -maxdepth 1 -type f -delete
+setup_asdf() {
+    if ! command -v asdf >/dev/null 2>&1; then
+        echo "Install asdf, before executing this script is required to manage scarb version"
+        exit 1
+    fi
+    asdf plugin-add scarb
+}
 
-# compile Cairo test contracts
-echo "Compiling Cairo contracts with $(poetry run starknet-compile-deprecated --version)"
+compile_contracts_with_scarb() {
+    CONTRACTS_DIRECTORY=$1
+    SCARB_WITH_VERSION=$(cat "$CONTRACTS_DIRECTORY/.tool-versions")
```

Comment:
useless use of cat

Reply:
I'm not sure how you want to do that, sth like:
```
SCARB_WITH_VERSION=$(<"$CONTRACTS_DIRECTORY/.tool-versions")
```

Reply:
```
SCARB_WITH_VERSION="${CONTRACTS_DIRECTORY}/.tool-versions"
```

Reply:
Maybe I miss sth but I want to take the content of this `.tool-versions` file 
```
    SCARB_WITH_VERSION_A=$(<"$CONTRACTS_DIRECTORY/.tool-versions")
    SCARB_WITH_VERSION="$CONTRACTS_DIRECTORY/.tool-versions"
    echo "SCARB_WITH_VERSION_A"
    echo $SCARB_WITH_VERSION_A
    echo "SCARB_WITH_VERSION"
    echo $SCARB_WITH_VERSION
```
returns

```
SCARB_WITH_VERSION_A
scarb 2.6.3
SCARB_WITH_VERSION
starknet_py/tests/e2e/mock/contracts_v2/.tool-versions
```

Reply:
sure, in that case:

```
SCARB_VERSION=$(awk '/scarb/ {print $2}' ${CONTRACTS_DIRECTORY}/.tool-versions)
```

should work

Reply:
this prints just a version, if you need to have the whole `scarb x.y.z` string, just print $0 in awk

Reply:
version is enough, thx!